deliver/lib/deliver/app_trailer.rb

```diff
@@ -0,0 +1,100 @@
+module Deliver
+  # AppTrailer represents one video trailer for one specific locale and
+  # device type.
+  class AppTrailer
+
+    attr_accessor :device_type
+
+    attr_accessor :video_path
+
+    attr_accessor :language
+
+    attr_accessor :timestamp
+
+    attr_accessor :poster_image_path
+
+    # @param video_path (String) path to the video file
+    # @param language (String) Language of this trailer (e.g. en-US)
+    def initialize(video_path, language)
+      self.video_path = video_path
+      self.language = language
+
+      # should be given as video file name prefix since we can't determine device by resolution
+      self.device_type = self.class.discover_device_type(video_path)
```

Comment:
just notice now. There's very little code in spaceship that uses the `self.class` code style. Could be replaced for consistency.